model/task_lifecycle.go

```diff
@@ -541,8 +544,11 @@ func UpdateUnblockedDependencies(t *task.Task) error {
 		if err = UpdateUnblockedDependencies(&blockedTask); err != nil {
 			return errors.WithStack(err)
 		}
-		if err = build.SetCachedTaskBlocked(blockedTask.BuildId, blockedTask.Id, blockedTask.Blocked()); err != nil {
-			return errors.Wrap(err, "error setting cached task unblocked")
+
+		if !blockedTask.IsPartOfDisplay() { // execution tasks not cached in build so we should skip
```

Comment:
if it's part of a display task and it's unblocked the display task might have been unblocked

Reply:
I think I was wrong about this too: the execution task that's been restarted is a dependency for the display task too, so we will call this for the display task.